src/component/addeditventure/index.tsx

```diff
@@ -1,72 +1,82 @@
-import { useContext } from "react";
-import { Controller } from "react-hook-form";
-import { useLocation, useNavigate, useParams } from "react-router-dom";
+import { useContext, useEffect } from "react";
+import { Controller, useForm } from "react-hook-form";
+import { useNavigate, useParams } from "react-router-dom";
 
 import TextField from "component/inputtext";
 import {
   DefaultAddEditVentureProps,
   PlasmicAddEditVenture,
 } from "component/plasmic/shared/PlasmicAddEditVenture";
 import { AuthContext } from "context/AuthContext";
-import { ventureNameError } from "module/errors";
+import { descriptionError, ventureNameError } from "module/errors";
 import { makeVentureUrl } from "module/helpers";
 import { useCreateVenture, useUpdateVenture } from "module/hook/venture";
 import { IVenture } from "module/interface/venture";
 
 interface AddEditVentureProps extends DefaultAddEditVentureProps {
   setIsActive: any;
-  currentVenture: IVenture;
+  currentVenture?: IVenture;
   hasVentures: boolean;
-  handleSubmit: any;
-  register: any;
-  control: any;
-  reset: any;
-  errors: any;
+  onChange?: (data: FormData) => void;
 }
 
+export type FormData = {
+  ventureName: string;
+  ventureDescription: string;
+};
+
 function AddEditVenture(props: AddEditVentureProps) {
+  const { setIsActive, currentVenture, hasVentures, onChange, ...rest } = props;
+
   const {
-    setIsActive,
     handleSubmit,
-    register,
-    reset,
-    errors,
-    currentVenture,
-    hasVentures,
     control,
-    ...rest
-  } = props;
+    watch,
+    reset,
+    formState: { errors },
+  } = useForm<FormData>({
+    defaultValues: {
+      ventureName: currentVenture?.name || "",
+      ventureDescription: currentVenture?.desc || "",
+    },
+  });
+
+  const values = watch();
+  useEffect(() => {
+    onChange && onChange(values);
+  }, [values, onChange]);
+
   const { ventureSlug } = useParams();
   const { token } = useContext(AuthContext);
   const { mutate: createVenture } = useCreateVenture();
   const { mutate: updateVenture } = useUpdateVenture();
 
   const navigate = useNavigate();
   const venture = currentVenture;
-  const url = useLocation();
   const handle = currentVenture?.name?.toLowerCase().replace(/\s/g, "");
   const ventureId = currentVenture?.id;
   const isEdit = ventureSlug && ventureSlug === handle ? "isEdit" : undefined;
 
-  const handleCreate = (data: any) => {
-    if (!data.ventureName) return;
+  const handleCreate = (data: FormData) => {
+    if (!data.ventureName || !data.ventureDescription) return;
+    const newHandle = data.ventureName.toLowerCase().replace(/\s/g, "");
```

Comment:
Fixes a bug I found where changing the name of a venture and saving redirects you to the old handle which is no longer valid.